core/java/iesi-core/src/main/java/io/metadew/iesi/datatypes/text/TextService.java

```diff
@@ -38,6 +38,19 @@ public Text resolve(String input, ExecutionRuntime executionRuntime) {
         return new Text(executionRuntime.resolveVariables(input));
     }
 
+    @Override
+    public boolean equals(Text _this, Text other, ExecutionRuntime executionRuntime) {
+        if (_this.getString() == null && other.getString() == null) {
+            return true;
+        } if (_this.getString() == null && other.getString() != null) {
+            return false;
+        } else if (_this.getString() != null) {
+            return _this.getString().equals(other.getString());
+        } else {
+            return false;
+        }
+    }
```

Comment:
```this.getString() == null || other.getString() == null```
will fail even if both are null. If both are null, equals should succeed

Reply:
If both are null, then it's catch by the first condition and return true,
else if (at this stage one of them isn't null) one of them is null, return false (because they aren't both null)
else perform equal